runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java

```diff
@@ -327,18 +328,15 @@ public Job translate(List<DataflowPackage> packages) {
       // not enabled.
       if (options.isEnableStreamingEngine()) {
         List<String> experiments = options.getExperiments();
-        if (experiments == null) {
-          experiments = new ArrayList<String>();
-        } else {
-          experiments = new ArrayList<String>(experiments);
-        }
         if (!experiments.contains(GcpOptions.STREAMING_ENGINE_EXPERIMENT)) {
           experiments.add(GcpOptions.STREAMING_ENGINE_EXPERIMENT);
         }
         if (!experiments.contains(GcpOptions.WINDMILL_SERVICE_EXPERIMENT)) {
           experiments.add(GcpOptions.WINDMILL_SERVICE_EXPERIMENT);
         }
-        options.setExperiments(experiments);
+        experiments
+            .parallelStream()
```

Comment:
parallel stream? I don't think we need to use multiple threads to do this.

Also, streams are [significantly slower](https://medium.com/@milan.mimica/slow-like-a-stream-fast-like-a-loop-524f70391182) then for loops even in JDK 12. You need to have 100k+ elements for parallel stream to beat stream and millions for parallel stream to beat for loop.